api/net/ip4/addr.hpp

```diff
@@ -107,10 +107,9 @@ struct Addr {
 
     const static std::regex ipv4_address_pattern
     {
-      "^\\s*(25[0–5]|2[0–4]\\d|[01]?\\d\\d?)\\."
-        "(25[0–5]|2[0–4]\\d|[01]?\\d\\d?)\\."
-        "(25[0–5]|2[0–4]\\d|[01]?\\d\\d?)\\."
-        "(25[0–5]|2[0–4]\\d|[01]?\\d\\d?)\\s*$"
+#define OCTET_PATTERN "(25[0-5]|2[0-4]\\d|[01]?\\d\\d?)"
+      "^\\s*" OCTET_PATTERN "\\." OCTET_PATTERN "\\." OCTET_PATTERN "\\." OCTET_PATTERN "\\s*$",
```

Comment:
Is that a trailing comma(**,**) I'm seeing there?

Although it's calling the second constructor on [this](http://en.cppreference.com/w/cpp/regex/basic_regex/basic_regex) list, it's better off not being there...